server.go

```diff
@@ -74,10 +95,13 @@ func NewServer(host, port string) *Server {
 	router.KeepContext = true
 
 	s := &Server{
-		addr:   host + ":" + port,
-		Router: router,
+		addr:     host + ":" + port,
+		Router:   router,
+		RequestIDFactory:     NewRequestIDFactory(),
```

Comment:
1) Why is this field public? Do we expect users to read / write it?
2) Why is the function `NewRequestIDFactory` public? Do we expect users to use it directly?